pkg/kwokctl/cmd/start/cluster/cluster.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"sigs.k8s.io/kwok/pkg/config"
+	"sigs.k8s.io/kwok/pkg/kwokctl/runtime"
+	"sigs.k8s.io/kwok/pkg/log"
+	"sigs.k8s.io/kwok/pkg/utils/path"
+)
+
+type flagpole struct {
+	Name       string
+	Components []string
+	Wait       time.Duration
+}
+
+// NewCommand returns a new cobra.Command for start cluster
+func NewCommand(ctx context.Context) *cobra.Command {
+	flags := &flagpole{}
+	cmd := &cobra.Command{
+		Args:  cobra.NoArgs,
+		Use:   "cluster",
+		Short: "Start a cluster or cluster component(s)",
+		Long:  "Start a cluster or cluster component(s)",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			flags.Name = config.DefaultCluster
+			return runE(cmd.Context(), flags)
+		},
+	}
+
+	cmd.PersistentFlags().StringArrayVar(&flags.Components, "component", flags.Components, fmt.Sprintf("Component name, which can include one or more of %v", config.EnabledComponentsByDefault.List()))
+	cmd.Flags().DurationVar(&flags.Wait, "wait", 0, "Wait for the cluster to be ready")
+
+	return cmd
+}
+
+func runE(ctx context.Context, flags *flagpole) error {
+	name := config.ClusterName(flags.Name)
+	workdir := path.Join(config.ClustersDir, flags.Name)
+
+	logger := log.FromContext(ctx)
+	logger = logger.With("cluster", flags.Name)
+	ctx = log.NewContext(ctx, logger)
+
+	rt, err := runtime.DefaultRegistry.Load(ctx, name, workdir)
+	if err != nil {
+		return err
+	}
+
+	components := []string{}
+	for _, component := range flags.Components {
+		if !config.EnabledComponentsByDefault.Has(component) {
+			components = append(components, component)
+		}
+	}
+	if len(components) != 0 {
+		return fmt.Errorf("the cluster component(s) does not contain %q", components)
+	}
+
+	if len(flags.Components) != 0 {
+		for _, component := range flags.Components {
```

Comment:
```suggestion
	if len(flags.Components) != 0 {
		unknownComponents := slices.Filter(flags.Components, func(component string) bool {
			return !config.EnabledComponentsByDefault.Has(component)
		})
		if len(unknownComponents) != 0 {
			return fmt.Errorf("the cluster component(s) does not exist: %v", unknownComponents)
		}

		for _, component := range flags.Components {
```

Add the utils function in `sigs.k8s.io/kwok/pkg/utils/slices`

``` golang
// Filter returns a new slice containing all elements in the slice that satisfy the predicate f.
func Filter[S ~[]T, T any](s S, f func(T) bool) []T {
	out := make([]T, 0, len(s))
	for _, v := range s {
		if f(v) {
			out = append(out, v)
		}
	}
	return out
}
```